File: naive/479.second-max-of-array/Solution.go
package main

func main() {

}

/**
 * @param nums: An integer array
 * @return: The second max number in the array.
 */
func secondMax(nums []int) int {
	// write your code here
	max, secMax := nums[0], nums[1]
	if max < secMax {
		max, secMax = secMax, max
	}
	for i, value := range nums {
		if i > 1 {
			if max < value {
				secMax = max
				max = value
			} else if secMax < value {
				secMax = value
			}
		}
	}
	return secMax
}

File: naive/479.second-max-of-array/Solution.java
public class Solution {
    /**
     * @param nums: An integer array
     * @return: The second max number in the array.
     */
    public int secondMax(int[] nums) {
        // write your code here
        int max = nums[0];
        int secondMax = nums[1];
        if (max < secondMax){
            int tmp = max;
            max = secondMax;
            secondMax = tmp;
        }
        for (int i=2;i<nums.length;i++){
            if (max<nums[i]){
                secondMax = max;
                max=nums[i];
            }else if(nums[i]>secondMax){
                secondMax = nums[i];
            }
        }
        return secondMax;
    }
}